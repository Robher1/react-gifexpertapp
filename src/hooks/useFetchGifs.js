import { useEffect, useState } from "react"
import { getGits } from '../helpers/getGifs';

export const useFetchGifts = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        
        getGits( category )
            .then( imgs => {
                setState({
                    data: imgs,
                    loading: false
                })
            } );
                
    }, [category])

    return state;
    
}