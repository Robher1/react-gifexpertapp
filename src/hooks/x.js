

function elJona( x ){
    const area = x*2/3;
    return area;
}

const elAreaDeJona = ( x ) => {
    const area = x*2/3;
    return area;
};

// fórmula para el área lateral de superficie de un cilindro es L. S. A. = 2 π rh.
function areaCilindro( radio, altura ){
    const area = 2 * Math.PI * radio * altura;
    return area;
}

console.log( areaCilindro( 3, 9 ) + 'in²' );