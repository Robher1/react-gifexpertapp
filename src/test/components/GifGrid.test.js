import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifts } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); // infigir llamadas

describe('Pruebas en <GifGrid />', ()=>{
    
    const category = 'mangos';
    
    test('debe de mostrar el componente correctamente', () => {

        useFetchGifts.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', async () => {

        const gifs = [{
            id: 'ABC',
            title: 'cosa',
            url: 'https://fake.jpg'
        },{
            id: '123',
            title: 'cosa',
            url: 'https://fake.jpg'
        }];

        useFetchGifts.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
    });
    

});