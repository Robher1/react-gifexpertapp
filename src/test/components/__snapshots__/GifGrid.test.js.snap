// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__lightSpeedInRight"
  >
    mangos
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__lightSpeedInRight"
  >
    mangos
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cosa"
      url="https://fake.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cosa"
      url="https://fake.jpg"
    />
  </div>
</Fragment>
`;
